[DDP][Instrumentation] Profiling range for bucket copy

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65772
* #65771
* #65770
* __->__ #65769

Seeing some bottlenecks when copying bucket to grad, help make it more
clear here.

Differential Revision: [D31217340](https://our.internmc.facebook.com/intern/diff/D31217340/)

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @gcramer23

rohan-varma committed:
[DDP][Instrumentation] Profiling range for bucket copy

Seeing some bottlenecks when copying bucket to grad, help make it more
clear here.

Differential Revision: [D31217340](https://our.internmc.facebook.com/intern/diff/D31217340/)

[ghstack-poisoned]

wayi1 reviewed (approved):
Have you checked if this will cause any non-trivial overhead? Better to confirm no performance regression before the submission.

rohan-varma commented:
I believe the codepath to optimize record_function when there are no profiling callbacks available has been optimized quite a bit, but yes will run a benchmark before landing.

rohan-varma committed:
Update on "[DDP][Instrumentation] Profiling range for bucket copy"


Seeing some bottlenecks when copying bucket to grad, help make it more
clear here.

Differential Revision: [D31217340](https://our.internmc.facebook.com/intern/diff/D31217340/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang gcramer23

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP][Instrumentation] Profiling range for bucket copy"


Seeing some bottlenecks when copying bucket to grad, help make it more
clear here.

Differential Revision: [D31217340](https://our.internmc.facebook.com/intern/diff/D31217340/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang gcramer23

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP][Instrumentation] Profiling range for bucket copy"


Seeing some bottlenecks when copying bucket to grad, help make it more
clear here.

Differential Revision: [D31217340](https://our.internmc.facebook.com/intern/diff/D31217340/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang gcramer23

[ghstack-poisoned]

mrshenli commented on torch/csrc/distributed/c10d/reducer.cpp:
```diff
@@ -1425,6 +1425,9 @@ void Reducer::finalize_bucket_dense(Bucket& bucket) {
     }
 
     if (!gradient_as_bucket_view_) {
+    RECORD_FUNCTION(
```
wrong indent?

rohan-varma committed:
Update on "[DDP][Instrumentation] Profiling range for bucket copy"


Seeing some bottlenecks when copying bucket to grad, help make it more
clear here.

Differential Revision: [D31217340](https://our.internmc.facebook.com/intern/diff/D31217340/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang gcramer23

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP][Instrumentation] Profiling range for bucket copy"


Seeing some bottlenecks when copying bucket to grad, help make it more
clear here.

Differential Revision: [D31217340](https://our.internmc.facebook.com/intern/diff/D31217340/)

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang gcramer23

[ghstack-poisoned]
